src/components/directed-requester.js

```diff
@@ -0,0 +1,42 @@
+const Requester = require('./requester');
+
+module.exports = class DirectedRequester extends Requester {
+    constructor(advertisement, discoveryOptions) {
+        super(advertisement, discoveryOptions);
+        this.sock.send = this.socketSend.bind(this);
+    }
+
+    socketSend(...args) {
+        // Enqueue if no socks connected yet
+        const requesterSock = this.sock;
+        if (!requesterSock.socks || !requesterSock.socks.length) return this.sock.enqueue(args);
```

Comment:
the original code has a debug statement here, and we'd be missing it, breaking the behavior. Could you add it?